file-formats/swcr/type/zif_aff_swcr_v1.intf.abap

```diff
@@ -10,57 +10,52 @@ INTERFACE zif_aff_swcr_v1
       abap_language_version TYPE zif_aff_types_v1=>ty_abap_language_version,
     END OF ty_header.
 
-  "! <p class="shorttext">Relation Type</p>
-  "! Relation Type
-  "! $values {@link zif_aff_swcr_v1.data:co_relation_type}
-  TYPES ty_relation_type TYPE c LENGTH 1.
+  TYPES:
+    "! <p class="shorttext">Access Permission</p>
+    "! Access Permission
+    BEGIN OF ty_permission,
+      "! <p class="shorttext">Software Component</p>
+      "! Software Component
+      "! $required
+      software_component TYPE c LENGTH 30,
+    END OF ty_permission.
 
-  CONSTANTS:
-    "! <p class="shorttext">Relation Type</p>
-    "! Relation Type
-    BEGIN OF co_relation_type,
-      "! <p class="shorttext">Unrestricted Usage</p>
-      "! Unrestricted Usage
-      unrestricted                 TYPE ty_relation_type VALUE '1',
-      "! <p class="shorttext">Key User Extensibility Usage</p>
-      "! Key User Extensibility Usage
-      key_user_extensibility_usage TYPE ty_relation_type VALUE '2',
-    END OF co_relation_type.
+  "! <p class="shorttext">Access Permissions</p>
+  "! Access Permissions
+  TYPES ty_permissions TYPE STANDARD TABLE OF ty_permission WITH DEFAULT KEY.
 
   TYPES:
-    "! <p class="shorttext">Related Software Component</p>
-    "! Related Software Component
-    BEGIN OF ty_relation,
-      "! <p class="shorttext">Related Software Component</p>
-      "! Related Software Component
+    "! <p class="shorttext">Dependency</p>
+    "! Dependency
+    BEGIN OF ty_dependency,
+      "! <p class="shorttext">Software Component</p>
+      "! Software Component
       "! $required
       software_component TYPE c LENGTH 30,
-      "! <p class="shorttext">Relation Type</p>
-      "! Relation Type
-      "! $required
-      type               TYPE ty_relation_type,
-    END OF ty_relation.
+    END OF ty_dependency.
 
-  "! <p class="shorttext">Relations</p>
-  "! Relations
-  TYPES ty_relations TYPE SORTED TABLE OF ty_relation WITH UNIQUE KEY software_component.
+  "! <p class="shorttext">Dependencies</p>
+  "! Dependencies
+  TYPES ty_dependencies TYPE STANDARD TABLE OF ty_dependency WITH DEFAULT KEY.
 
   TYPES:
     "! <p class="shorttext">Software Component Relations</p>
     "! Software Component Relations
     BEGIN OF ty_main,
       "! <p class="shorttext">Format Version</p>
-      "! Format version
+      "! Format Version
       "! $required
       format_version TYPE zif_aff_types_v1=>ty_format_version,
       "! <p class="shorttext">Header</p>
       "! Header
       "! $required
       header         TYPE ty_header,
-      "! <p class="shorttext">Relations</p>
-      "! Relations
-      "! $required
-      relations      TYPE ty_relations,
+      "! <p class="shorttext">Access Permissions</p>
+      "! The Software Component Grants Access Permission to the Specified Software Components
+      permissions    TYPE ty_permissions,
```

Comment:
I am just wondering, whether we should keep the short text and the attribute name in sync. I guess you know better whether "access permissions" or just "permissions" is the better term.

Reply:
We would like to keep the "permissions" term for the attribute.

Reply:
One more Would you then also suggest to name the shorttext just "Permission"? Or is there any reason why you don't want to keep the shorttext and the field name in sync?

Reply:
For the time being, all supported types of permissions are access permissions. Therefore we want to keep the shorttext as specific and understandable as possible, i.e. we would like to stay with "Access Permissions". As soon as there are use cases for other types of permissions we would adjust the shorttext accordingly.

Reply:
OK, I understand. Thanks for the clarification